accepted/00076-configuration-channels-visibility.md

```diff
@@ -0,0 +1,84 @@
+- Feature Name: Enhance visibility of connected Configuration/State Channels
+- Start Date: 2020-12-17
+- RFC PR: 
+
+# Summary
+[summary]: #summary
+
+Provide visibility and overview of **inherited** Configuration and State Channels connections at a System Details level. The reverse information as well should be provided by giving visibility of those systems they are connected (directly or by **inheritance**) to a certain Configuration or State Channel at a Channel Details level.
+
+*Note: everything discussed in this RFC applies for Salt clients only. Traditional clients do not inherit Configuration channel assignments from System Groups/Organizations.*
+
+
+# Motivation
+[motivation]: #motivation
+
+### Why are we doing this?
+
+At the moment there is no way to figure out Configuration and State Channels connections other than the ones directly in place to a specific System. Such a type of channel can exist and be defined at:
+
+- Organization level
+- System Group level
+- System level
+
+but only the third level is visible for each System. This means that once Salt States are applied, there will be more than what is visible: Salt States are a set of defined states from
+
+Organization Configuration Channels **+** System Group Configuration Channels **+** System Configuration Channels
+
+### What use cases does it support?
+
+Typical and complete use case:
+
+1. create an Organization O1
+2. create a System Group G1
+3. onboard a System S1 as a Salt client in the Organization O1 and System Group G1
+4. create a State Channel C1 at an Organization O1 level
+5. create a State Channel C2 at a System Group G1 level
+6. create a State Channel C3 directly at a System S1 detail level
+7. go to System Detail > States > Configuration Channels > System
+8. the C3 channel is listed as an assignment to S1
+9. there is no info about C1 nor C2 channels, but because of step 4. 5. in reality S1 will receive not only C3 but also C1 and C2 as inheritance of its own setup, being part of O1 and G1
+10. the reverse logic is valid as well, there is no information about systems assigned to channels C1 or C2 in Configuration > Channels > Channel Overview > Systems because. No system is directly assigned to them, but S1 inherits them.
+
+Additional use case: a system can be part of multiple groups, therefore **inheritance** can become wide and complex really fast.
```

Comment:
Rather than qualitatively explained ("become wide and complex really fast") such complexity should be explained here IMO.

How is inheritance defined in this context (pointer to existing documentation if any)? What are the elements of such complexity? What does it mean it increases in a "really fast" way?

Remember that the aim of the RFC is not to sell the feature - rather, to explain the problem as an introduction to the solution, which is the part under scrutiny here. The intent is to submit the solution to peers before going to implementation to reduce the risk of "bad surprises" at PR time.

Reply:
> Rather than qualitatively explained ("become wide and complex really fast") such complexity should be explained here IMO.
> 
> How is inheritance defined in this context (pointer to existing documentation if any)? What are the elements of such complexity? What does it mean it increases in a "really fast" way?

Probably "complexity" is not the right word then. I left out the detailed explanation of such a "wide and complex scenario" because it is still the original problem explained. It is only about the same **inheritance invisibility**, but multiplied for every group a system can be part of.
If you are part of one group, you inherit from that group only: in such a case you can even deduct where do they comes from. If they are directly assigned, you got them. If they are not directly assigned, well they come from the single group you are in.
But if you are part of 5 different groups, and you inherit from all of them, the resulting set will be uncategorized and you cannot distinguish what comes from group G1, group G2, etc.

I'm adding here two pictures as they usually give more insight than many many words. I plan to inject them into the RFC, anything against or anything you would like to change about it?

### Visible inheritance
![visible-inheritance](https://user-images.githubusercontent.com/7080830/104619161-d773cc80-568d-11eb-9589-ea7e7141780e.png)

### Hidden inheritance result
![uncategorized-result](https://user-images.githubusercontent.com/7080830/104619197-e35f8e80-568d-11eb-8715-e15dffd8c3c6.png)




> 
> Remember that the aim of the RFC is not to sell the feature - 

JFYI, I don't see any marketing slogan here to sell the feature actually :/

Reply:
As per our meeting, this PR is to be considered WIP for now, so let's just suspend any reviewing until it's ready.